module/VuFind/src/VuFind/Controller/HoldsController.php

```diff
@@ -147,10 +155,26 @@ public function listAction()
 
             $driversNeeded[] = $current;
         }
+        // Cache the current list of requests for editing:
+        $this->putCachedData(
+            $this->getCacheId($patron, 'holds'),
+            $driversNeeded,
+            600
+        );
 
         // Get List of PickUp Libraries based on patron's home library
         try {
-            $view->pickup = $catalog->getPickUpLocations($patron);
+            $view->pickup = $this->getCachedData(
+                $this->getCacheId($patron, 'pickup')
```

Comment:
Is it worth putting the cache key in a variable since we calculate it twice close together here and in line 173?

Reply:
Done.